packages/-ember-data/addon/setup-container.js

```diff
@@ -46,6 +51,22 @@ function initializeStore(application) {
   }
 }
 
+// Implicit injection was removed. This is a replacement for Ember Route implicit store
+// https://github.com/emberjs/rfcs/pull/774
+Object.defineProperty(Route, 'store', {
```

Comment:
This would define a property on the `Route` class itself, right? Why not use `reopen`?

Reply:
You were right.  I has thought defineProperty and reopen would be equivalent but turns out I was wrong.

Reply:
I'm guessing this has something to do with EmberObject.  But overall I was trying the other way to avoid the `reopen` deprecation